tests/stp_test.py

```diff
@@ -0,0 +1,703 @@
+Name             rity Cost      Fast Fast                 Cost        Root             Bridge
+Ethernet4        128  200       N    N      FORWARDING    400         0064b86a97e24e9c 806480a235f281ec
+"""
+
+show_spanning_tree_statistics = """\
+VLAN 100 - STP instance 0
+--------------------------------------------------------------------
+PortNum          BPDU Tx        BPDU Rx        TCN Tx         TCN Rx
+Ethernet4        10             15             15             5
+"""
+
+show_spanning_tree_bpdu_guard = """\
+PortNum          Shutdown     Port Shut
+                 Configured   due to BPDU guard
+-------------------------------------------
+Ethernet4        No           NA
+"""
+
+show_spanning_tree_root_guard = """\
+Root guard timeout: 30 secs
+
+Port             VLAN   Current State
+-------------------------------------------
+Ethernet4        100    Consistent state
+"""
+
+
+class TestStp(object):
+    @classmethod
+    def setup_class(cls):
+        os.environ['UTILITIES_UNIT_TESTING'] = "1"
+        print("SETUP")
+
+    def test_show_spanning_tree(self):
+        runner = CliRunner()
+        db = Db()
+        result = runner.invoke(show.cli.commands["spanning-tree"], [], obj=db)
+        print(result.exit_code)
+        print(result.output)
+        assert result.exit_code == 0
+        assert result.output == show_spanning_tree
+
+    def test_show_spanning_tree_vlan(self):
+        runner = CliRunner()
+        db = Db()
+        result = runner.invoke(show.cli.commands["spanning-tree"].commands["vlan"], ["100"], obj=db)
+        print(result.exit_code)
+        print(result.output)
+        assert result.exit_code == 0
+        assert result.output == show_spanning_tree_vlan
+
+    def test_show_spanning_tree_statistics(self):
+        runner = CliRunner()
+        db = Db()
+        result = runner.invoke(show.cli.commands["spanning-tree"].commands["statistics"], [], obj=db)
+        print(result.exit_code)
+        print(result.output)
+        assert result.exit_code == 0
+        assert result.output == show_spanning_tree_statistics
+
+    def test_show_spanning_tree_statistics_vlan(self):
+        runner = CliRunner()
+        db = Db()
+        result = runner.invoke(
+            show.cli.commands["spanning-tree"].commands["statistics"].commands["vlan"], ["100"], obj=db)
+        print(result.exit_code)
+        print(result.output)
+        assert result.exit_code == 0
+        assert result.output == show_spanning_tree_statistics
+
+    def test_show_spanning_tree_bpdu_guard(self):
+        runner = CliRunner()
+        db = Db()
+        result = runner.invoke(show.cli.commands["spanning-tree"].commands["bpdu_guard"], [], obj=db)
+        print(result.exit_code)
+        print(result.output)
+        assert result.exit_code == 0
+        assert result.output == show_spanning_tree_bpdu_guard
+
+    def test_show_spanning_tree_root_guard(self):
+        runner = CliRunner()
```

Comment:
Consider using fixtures since these are shared by all testcases. This enforces every future testcase to run throught the same setup.

Reply:
Done